src/Actions/Component/UpdateComponent.php

```diff
@@ -10,11 +11,13 @@ class UpdateComponent
 {
     use AsAction;
 
-    public function handle(Component $component, array $data): Component
+    public function handle(Component $component, ComponentData $data): Component
     {
         $oldStatus = $component->status;
 
-        $component->update($data);
+        $component->update(array_filter(
+            $data->toArray(),
+        ));
```

Comment:
Not too happy about the use of `array_filter` here. This assumes that you can't set a column to `null`. Might have to think about this a bit more…